src/vs/workbench/api/node/mainThreadHandlerRegistry.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+/**
+ * MainThreadHandlerRegistry maintains a collection of objects by handle.
+ */
+export class MainThreadHandlerRegistry<T> {
+
+	private _nextChildHandle: number = 0;
+	private _registrations: { [handle: number]: { [childHandle: number]: T }; } = Object.create(null);
```

Comment:
I new to Typescript and ES6 so I don't have a strong reason other than I copied this style from elsewhere in the VS Code codebase (trying to be consistent) and the fact that I can label the keys/values seems nice. Do you think I should use a Map?